.circleci/config.yml

```diff
@@ -1,8 +1,166 @@
+                DEPLOY_ENV=dev
+              ;;
+              test)
+                DEPLOY_ENV=test
+                NG_BUILD_OPTS=--prod --aot
+              ;;
+              staging)
+                DEPLOY_ENV=staging
+                NG_BUILD_OPTS=--prod --aot
+              ;;
+              staging)
+                DEPLOY_ENV=prod
+                NG_BUILD_OPTS=--prod --aot
+              ;;
+              *)
+                echo "Cannot map << pipeline.git.branch >> to a deployment environment"
+                exit 17
+              ;;
+            esac
+            echo "Setting deployment ENVIRONMENT to $DEPLOY_ENV"
+            echo "export ENVIRONMENT=$DEPLOY_ENV" >> $BASH_ENV
+            echo 'export NG_BUILD_OPTS=$NG_BUILD_OPTS' >> $BASH_ENV
+            source $BASH_ENV
+      - run:
+          name: <<parameters.study_key>> show environment variables
+          command: env
+      - run:
+          name: <<parameters.study_key>> generate config files
+          command: ./build-study.sh v1 $ENVIRONMENT <<parameters.study_key>> <<parameters.study_guid>> --config
+          environment:
+            USE_DOCKER: false
+      # todo: conditionally don't build if we already have cached the tarball for the SHA+ENV somewhere (circle artifacts?  github package manager?  GCP bucket?
+      # for better build reproducibility, we want to move the same build artifacts to different environments and not rebuild them and
+      # suffer the pain of npm caching, package-lock, and auto-upgrade pitfalls
+      - restore_cache:
+          key: *npm_cache_key
+      - run:
+          name: <<parameters.study_key>> npm install
+          command: npm install
+          working_directory: ~/repo/ddp-workspace
+      - save_cache:
+          key: *npm_cache_key
+          paths:
+            - ./node_modules
+      - run:
+          name: <<parameters.study_key>> ng build
+          command: ng build $ANGULAR_PROJECT_NAME $NG_BUILD_OPTS --base-href=/ --output-path=$ANGULAR_PROJECT_DIR_PATH/dist --verbose=true --progress=true
+          working_directory: ~/repo/ddp-workspace
+      - run:
+          name: <<parameters.study_key>> setup gcloud context
+          command: |
+            echo $GCLOUD_SERVICE_KEY | gcloud auth activate-service-account --key-file=-
+            # ran out of time, but demonstration of basic idea extracting data from Vault directly. Needs to be wrapped in nicer way!
+            #gcloud --quiet config set compute/zone `vault read  -format=json  secret/pepper/$DEPLOY_ENV/v1/conf | jq -r '.data.gcp.computeZone'`
+      - run:
+          name: <<parameters.study_key>> deploy to GAE
+          command: gcloud app deploy --version="${CIRCLE_SHA1}" --project "broad-ddp-${ENVIRONMENT}" "${ANGULAR_PROJECT_DIR_PATH}/app.yaml"
+          working_directory: ~/repo/ddp-workspace
+
+
 jobs:
-  build:
-    docker: 
-      - image: circleci/node:4.8.2 # the primary container, where your job's commands are run
+  osteo:
```

Comment:
These jobs now run in parallel.  To add a new study, just copy-paste a few lines and set `study_key` and  `study_guid`.